src/presentation/http/router/note.test.ts

```diff
@@ -518,6 +518,221 @@ describe('Note API', () => {
+        creatorId: user.id,
+      });
+
+      /** Create test note settings */
+      await global.db.insertNoteSetting({
+        noteId: childNote.id,
+        isPublic: true,
+      });
+
+      /** Create test note relation */
+      await global.db.insertNoteRelation({
+        parentId: parentNote.id,
+        noteId: childNote.id,
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${childNote.publicId}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      expect(response?.json()).toMatchObject({
+        parents: [
+          {
+            id: parentNote.publicId,
+            content: parentNote.content,
+          },
+          {
+            id: childNote.publicId,
+            content: childNote.content,
+          },
+        ],
+      });
+    });
+
+    test('Returns two parents note in case when note has two parents', async () => {
+      /** Create test user */
+      const user = await global.db.insertUser();
+
+      /** Create acces token for the user */
+      const accessToken = global.auth(user.id);
+
+      /** Create test note - a grand parent note */
+      const grandParentNote = await global.db.insertNote({
+        creatorId: user.id,
+      });
+
+      /** Create test note - a parent note */
+      const parentNote = await global.db.insertNote({
+        creatorId: user.id,
+      });
+
+      /** Create test note - a child note */
+      const childNote = await global.db.insertNote({
+        creatorId: user.id,
+      });
+
+      /** Create test note settings */
+      await global.db.insertNoteSetting({
+        noteId: childNote.id,
+        isPublic: true,
+      });
+
+      /** Create note relation between parent and grandParentNote */
+      await global.db.insertNoteRelation({
+        parentId: grandParentNote.id,
+        noteId: parentNote.id,
+      });
+
+      /** Create test note relation */
+      await global.db.insertNoteRelation({
+        parentId: parentNote.id,
+        noteId: childNote.id,
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${childNote.publicId}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      expect(response?.json()).toMatchObject({
+        parents: [
+          {
+            id: grandParentNote.publicId,
+            content: grandParentNote.content,
+          },
+          {
+            id: parentNote.publicId,
+            content: parentNote.content,
+          },
+          {
+            id: childNote.publicId,
+            content: childNote.content,
+          },
+        ],
+      });
+    });
+
+    test('Returns one parent\'s note when the user is not in the parent note\'s team but shares a relation', async () => {
```

Comment:
```suggestion
    test('Returns one parent note when the user is not in the parent note\'s team but shares a relation', async () => {
```

Reply:
also if note has parent, that already means that relation exists, so `but shares a relation` is unneded

Reply:
also, i did not get the case, if one parent note is returned, then overall two notes would be returned? (actual note and parent) or only just actual note?

Reply:
In this case, the note is not created by the same user, at the same time the main user does not exist in the team of the parent note. Just to prove that the notes are returned based on the note relation, not on team members of the note.